uniffi_macros/src/export/scaffolding.rs

```diff
@@ -114,27 +135,41 @@ impl ScaffoldingBits {
             .chain(sig.scaffolding_params())
             .collect();
         let param_lifts = sig.lift_exprs();
+        let mut rust_fn_call = quote! { uniffi_self.#ident(#(#param_lifts,)*) };
+        if udl_mode && sig.looks_like_result {
+            rust_fn_call = quote! { #rust_fn_call.map_err(::std::convert::Into::into) }
+        }
+        let return_ffi_converter = sig.return_ffi_converter();
 
         Self {
             params,
             pre_fn_call: quote! {
-                let uniffi_self = #ffi_converter::try_lift(uniffi_self_lowered).unwrap_or_else(|err| {
-                    ::std::panic!("Failed to convert arg 'self': {}", err)
-                });
+                let uniffi_self = match #ffi_converter::try_lift(uniffi_self_lowered) {
+                    Ok(v) => v,
+                    Err(e) => return Err(#return_ffi_converter::handle_failed_lift("self", e)),
+                };
             },
-            rust_fn_call: quote! { uniffi_self.#ident(#(#param_lifts,)*) },
+            rust_fn_call,
         }
     }
 
-    fn new_for_constructor(sig: &FnSignature, self_ident: &Ident) -> Self {
+    fn new_for_constructor(sig: &FnSignature, self_ident: &Ident, udl_mode: bool) -> Self {
         let ident = &sig.ident;
         let params: Vec<_> = sig.args.iter().map(NamedArg::scaffolding_param).collect();
         let param_lifts = sig.lift_exprs();
+        let mut rust_fn_call = quote! { #self_ident::#ident(#(#param_lifts,)*) };
+        if udl_mode {
+            if sig.looks_like_result {
+                rust_fn_call = quote! { #rust_fn_call.map(::std::sync::Arc::new).map_err(::std::convert::Into::into) };
+            } else {
+                rust_fn_call = quote! { ::std::sync::Arc::new(#rust_fn_call) };
+            }
+        }
```

Comment:
No need to make this `mut`, can just move the initial value for `rust_fn_call` into an `else`.